server/odc-service/src/main/java/com/oceanbase/odc/service/integration/model/Oauth2Parameter.java

```diff
@@ -100,6 +99,11 @@ private Provider toProvider(String issueUrl) {
         return provider;
     }
 
+    @SneakyThrows
+    private String getRedirectHost(String redirectUrl) {
+        URL url = new URL(redirectUrl);
+        return url.getProtocol() + "://" + url.getHost() + (url.getPort() > 0 ? ":" + url.getPort() : "");
```

Comment:
if port is 80 for http protocal, or port is 443 for https protocal, skip  port part in url.

Reply:
may refer spring util `UrlUtils.buildFullRequestUrl`

Reply:
`UrlUtils.buildFullRequestUrl` is more applicable to Request objects, which returns the default port of the protocol when you get port. Here use java.net.URL to parse url string, if there is no port, the value of port will be -1, which will be lead error.